tests/benchdnn/dnnl_common.cpp

```diff
@@ -366,8 +366,40 @@ int execute_and_wait(perf_function_t &exec_func, const dnnl_engine_t &engine,
 
     execute_unmap_args(args, dnnl_args);
 
-    auto status = exec_func(stream, dnnl_args);
-    DNN_SAFE(dnnl_stream_wait(stream), CRIT);
+    dnnl_status_t status = dnnl_runtime_error;
+    bool run_regular_exec = true;
+#if DNNL_GPU_RUNTIME == DNNL_RUNTIME_DPCPP
+    while (use_sycl_graph && is_gpu(engine)) {
+        void *queue_ptr;
+        DNN_SAFE(dnnl_sycl_interop_stream_get_queue(stream, &queue_ptr), CRIT);
+        sycl::queue queue = *static_cast<sycl::queue *>(queue_ptr);
+        const bool can_run_sycl_graph = queue.get_device().get_backend()
+                == sycl::backend::ext_oneapi_level_zero;
+        if (!can_run_sycl_graph) break;
+
+        BENCHDNN_PRINT(
+                2, "%s\n", "[INFO] Using experimental SyCL graph execution.");
+        sycl::ext::oneapi::experimental::command_graph graph {
+                queue.get_context(), queue.get_device()};
+
+        graph.begin_recording(queue);
+        status = exec_func(stream, dnnl_args);
+        graph.end_recording(queue);
+        DNN_SAFE(dnnl_stream_wait(stream), CRIT);
+
+        auto exec = graph.finalize();
+        queue.ext_oneapi_graph(exec).wait();
```

Comment:
Can you give a bit more context on the reasons behind this PR in general? I see it is adding a way to test the SYCL Graph extension with benchdnn. Is the intention to test for correctness and/or measure the impact on performance? Wouldn't it be useful to also be able to re-use the graph `n` times?
Also the PR says it is an experimental support. Is there a plan for the future of SYCL-Graph in benchdnn?

Reply:
The main goal is enabling functional validation of oneDNN primitives in SYCL Graph record/replay mode. While use of SYCL Graph does not require changes in oneDNN API it puts some limitations on what the implementation can do internally.

'Experimental' refers to [support status](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc#5-status) of SYCL Graph extension support in the compiler.